docs/AA--Javascript-in-EE/asset-caching-by-filename.md

```diff
@@ -6,30 +6,76 @@ We also briefly considered using `filemtime` as the "version" generator for our
 
 So we've decided to [switch to filenames containing a hash](https://github.com/eventespresso/event-espresso-core/pull/287) generated from the content of the files.  This means if the content of a built file does not change, then the filename remains the same, if the content of a built bundle file changes, then the filename will as well.
 
-Switching to this new process requires some tweaks to how our assets are registered with `wp_register_script` however.
+Switching to this new process requires some tweaks to where assets are located, and how they are registered with `wp_register_script`.
+
+## Where Javascript and CSS Assets Should be Located
+
+In Event Espresso core, new Javascript and CSS files should be located somewhere under `assets/src` (previously they were placed in `core/templates/global_assets`, or beside the PHP files they corresponded to.)
+Put the files in whatever folder/subfolder is logical, like `assets/src/empire/stardestroyer.js'.
+`
+Then add an new entry to `webpack.common.js`' `config` const. Use whatever string you like for the entry slug/chunk name.
+Like the following
+
+```javascript
+...
+{
+		entry: {
+			'eventespresso-core-star-destroyer-chunkname' /* <!-- entry slug/chunk name! */: [
+				assets + 'empire/stardestroyer.js',
+			],
+		},
+		output: {
+			filename: '[name].[chunkhash].dist.js',
+			path: path.resolve( __dirname, 'assets/dist' ),
+		},
+		module: moduleConfigWithJsAndCssRules,
+		watchOptions: {
+			poll: 1000,
+		},
+	},
+...
+```
+You've now informed webpack where the new assets are located, and what their chunk name is.
+
+Don't forget to run the build process with `npm run watch` to build the files. 
+
+After that, the server-side PHP Registry code will know how to find them given the chunk name. 
 
 ## Registering a Event Espresso Core asset with a hash in its name.
```

Comment:
I know you didn't do this, but should be changed to:

> ## Registering an Event Espresso Core asset with a hash in its name.

Reply:
Oh man! We totally forgot about the "'a' becomes 'an' when the following word begins with a vowel". That new feature has been in English since at least version 1650!